kdwsdl2cpp/common/fileprovider.h

```diff
@@ -40,8 +43,18 @@ class KXMLCOMMON_EXPORT FileProvider : QObject
     bool get( const QUrl &url, QString &target );
     void cleanUp();
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
+    static bool loadCertificate(const QString & certPath, const QString &password = QString());
+#endif
+
   private:
     QString mFileName;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
+    static QSslKey mSslKey;
+    static QSslCertificate mCertificate;
+    static QList<QSslCertificate> mCaCertificates;
+    static bool mCertificateLoaded;
```

Comment:
I am not a fan of static members. Maybe you could use Settings to store this information?

Reply:
My motivation behind making these static was to simplify the certificate loading to the fileprovider. If the user pass the certificate I think (due to the nature of the application) it should be expectable to be used in all requests. By using the static members if any developer in the future utilise the Fileprovider the certs will be used automatically, it could not be forgotten. 

What is your objection about the static members? (I am just curious.)

Reply:
I think that the desired behaviour can be reached by using the Settings singleton. It holds most of the command line settings and I think pkcs12 should be one of them. It probably makes sense to move the load function as well.

I don't have a very concrete reason why, other then that I don't like mixing static and non-static data in the same class definition. My opinion is that a instance of a class should either be self-serving (not dependant on globals, only on references to other classes) or there should only be 1 instance of it (singleton). I am not sure that is a clear explanation. In this case I would prefer one singleton containing all settings and multiple FileProviders holding a reference to the settings (Settings::self() is used in KDSoap).

Reply:
> I don't have a very concrete reason why, other then that I don't like mixing static and non-static data in the same class definition. 

Thanks for the explanation, totally makes sense, I will fix it soon.